src/olympia/reviewers/tests/test_decorators.py

```diff
@@ -34,16 +32,18 @@ def test_301_by_slug(self):
         self.assert3xx(res, self.id_path, 301)
 
     def test_301_by_guid(self):
-        self.request.path = f'http://testserver/addon/{quote(self.addon.guid)}/reviews'
+        self.request.path = f'http://testserver/addon/{self.addon.guid}/reviews'
         res = self.view(self.request, self.addon.guid)
         self.assert3xx(res, self.id_path, 301)
 
     def test_slug_replace_no_conflict(self):
```

Comment:
what's this tests supposed to be checking for?  the slug doesn't seem to be replaced, and no conflict with what?

Reply:
If the slug is somehow present twice in the URL, we don't replace twice - said differently, we only replace the part of the URL that matters to look up the add-on.

As I've noted in the `FIXME` in the decorator (and in my comment above that rewords that `FIXME`) the whole replace  method, copied from the original `addons` decorator, is still not great and can lead to incorrect URLs depending on the slug, though.